src/transactions/PaymentOpFrame.cpp

```diff
@@ -125,16 +125,19 @@ PaymentOpFrame::doCheckValid(uint32_t ledgerVersion)
 }
 
 void
-PaymentOpFrame::insertLedgerKeysToPrefetch(
-    std::unordered_set<LedgerKey>& keys) const
+PaymentOpFrame::insertLedgerKeysToPrefetch(std::unordered_set<LedgerKey>& keys,
+                                           uint32_t ledgerVersion) const
 {
     keys.emplace(accountKey(mPayment.destination));
 
     // Prefetch issuer for non-native assets
     if (mPayment.asset.type() != ASSET_TYPE_NATIVE)
     {
-        auto issuer = getIssuer(mPayment.asset);
-        keys.emplace(accountKey(issuer));
+        if (ledgerVersion < 13)
```

Comment:
we should just stop pre-fetching issuers even for protocol 12: this avoids the code bloat related to passing `ledgerVersion` through the various layers

Reply:
Done